toolkit/tools/graphanalytics/graphanalytics.go

```diff
@@ -0,0 +1,295 @@
+		if node.Type != pkggraph.TypeRun && node.Type != pkggraph.TypeBuild {
+			continue
+		}
+
+		if node.State == pkggraph.StateUnresolved {
+			continue
+		}
+
+		dependentName := nodeDependencyName(node)
+
+		search := traverse.BreadthFirst{}
+		search.Walk(pkgGraph, node, func(n graph.Node, d int) (stopSearch bool) {
+			dependencyNode := n.(*pkggraph.PkgNode)
+
+			if dependencyNode.State == pkggraph.StateUnresolved {
+				unresolvedPkgName := dependencyNode.VersionedPkg.Name
+				insertIfMissing(unresolvedPackageDependents, unresolvedPkgName, dependentName)
+			}
+
+			return
+		})
+	}
+
+	printTitle("[INDIRECT] Most common unresolved dependencies")
+	printMap(unresolvedPackageDependents, "total dependents", maxResults)
+
+	return
+}
+
+// printDirectlyMostUnresolved will print the top unresolved packages that are directly most blocking.
+func printDirectlyMostUnresolved(pkgGraph *pkggraph.PkgGraph, maxResults int) {
+	unresolvedPackageDependents := make(map[string][]string)
+
+	for _, node := range pkgGraph.AllRunNodes() {
+		if node.State != pkggraph.StateUnresolved {
+			continue
+		}
+
+		pkgName := node.VersionedPkg.Name
+
+		dependents := pkgGraph.To(node.ID())
+		for dependents.Next() {
+			dependent := dependents.Node().(*pkggraph.PkgNode)
+
+			// Do not consider goal nodes
+			if dependent.Type == pkggraph.TypeGoal {
+				continue
+			}
+
+			dependentName := nodeDependencyName(dependent)
+			insertIfMissing(unresolvedPackageDependents, pkgName, dependentName)
+		}
+	}
+
+	printTitle("[DIRECT] Most common unresolved dependencies")
+	printMap(unresolvedPackageDependents, "direct dependents", maxResults)
+
+	return
+}
+
+// printDirectlyClosestToBeingUnblocked will print the packages with the fewest unresolved direct build requires.
+func printDirectlyClosestToBeingUnblocked(pkgGraph *pkggraph.PkgGraph, maxResults int) {
+	srpmsBlockedBy := make(map[string][]string)
+
+	for _, node := range pkgGraph.AllNodes() {
+		if node.Type != pkggraph.TypeBuild {
+			continue
+		}
+
+		if node.State == pkggraph.StateUnresolved {
+			continue
+		}
+
+		pkgSRPM := filepath.Base(node.SrpmPath)
+
+		dependencies := pkgGraph.From(node.ID())
+		for dependencies.Next() {
+			dependency := dependencies.Node().(*pkggraph.PkgNode)
+			// Only consider unresolved build nodes.
+			if dependency.State != pkggraph.StateBuild && dependency.State != pkggraph.StateUnresolved {
+				continue
+			}
+
+			// Skip requirements that come from the same srpm.
+			dependencySRPM := filepath.Base(dependency.SrpmPath)
+			if pkgSRPM == dependencySRPM {
+				continue
+			}
+
+			dependencyName := nodeDependencyName(dependency)
+			insertIfMissing(srpmsBlockedBy, pkgSRPM, dependencyName)
+		}
+	}
+
+	printTitle("[DIRECT] SRPMs closest to being ready to build")
+	printReversedMap(srpmsBlockedBy, "unmet dependencies", maxResults)
+}
+
+// printDirectlyClosestToBeingUnblocked will print the packages with the fewest unresolved indrect build requires.
```

Comment:
>printDirectlyClosestToBeingUnblocked will print the packages with the fewest unresolved indrect build requires. [](start = 3, length = 111)

This comment seems identical to the one, which prints things "directly". It might be worth to add a sentence explaining the difference.